### PR TITLE
json, nbt, command api improvement.

### PR DESCRIPTION
**Pull request by portlek**

### CODE REVIEW

**Comment by portlek** [comment]
finally, its finished now i think.

File: build.gradle [hunk 1]
```diff
@@ -18,6 +18,7 @@ dependencies {
     // For use with mods, should already be included in the template
     implementation 'com.electronwill.night-config:toml:3.4.0'
     implementation 'org.apache.logging.log4j:log4j-api:2.8.1'
+    implementation 'org.jetbrains:annotations:17.0.0'
```
Comment (Admicos) [inline build.gradle]: You might want to use `compileOnly` for annotations

File: src/main/java/club/issizler/okyanus/api/item/ItemStack.java [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+
+import club.issizler.okyanus.api.nbt.NBTBase;
+import club.issizler.okyanus.api.nbt.NBTEndOf;
+import org.jetbrains.annotations.NotNull;
+
+public class ItemStack {
+
+    @NotNull
+    public NBTBase getTag() {
+        return new NBTEndOf();
```
Comment (Admicos) [inline src/main/java/club/issizler/okyanus/api/item/ItemStack.java]: Shouldn't ItemStack be an interface, with an implementation in runtime?

File: src/main/java/club/issizler/okyanus/api/json/ClickFeature.java [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+    private final JsonEvent jsonEvent;
+
+    public ClickFeature(@NotNull final JsonEvent jsonEvent) {
+        this.jsonEvent = jsonEvent;
+    }
+
+    @NotNull
+    @Override
+    public String handle() {
+        return ",\"clickEvent\":{\"action\":\"" + jsonEvent.action() + "\",\"value\":" + jsonEvent.value() + '}';
```
Comment (Admicos) [inline src/main/java/club/issizler/okyanus/api/json/ClickFeature.java:16]: Honestly, you might want to use a proper JSON serialization library, like Gson (which should be included with Minecraft (or Fabric, can't remember), and thus free to use for us)

File: src/main/java/club/issizler/okyanus/api/json/JsonMessage.java [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+
+import club.issizler.okyanus.api.entity.Player;
+import org.jetbrains.annotations.NotNull;
+
+public interface JsonMessage {
+
+    /**
+     * Sends the message to online players
+     */
+    void sendAll();
```
Comment (Admicos) [inline src/main/java/club/issizler/okyanus/api/json/JsonMessage.java:11]: Are we absolutely sure these should be in the message? Is it the message's responsibility to send itself?

File: src/main/java/club/issizler/okyanus/api/json/click/RunConsumer.java [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+import static club.issizler.okyanus.api.cmdnew.CommandOptions.*;
+
+public final class RunConsumer implements JsonEvent {
+
+    private final UUID uuid = UUID.randomUUID();
+    private final Command command;
+
+    public RunConsumer(boolean removeAfter, @NotNull final Proc<Player> proc) {
+        command = new CommandOf(
+            "uuid",
```
Comment (Admicos) [inline src/main/java/club/issizler/okyanus/api/json/click/RunConsumer.java:24]: Wouldn't having multiple commands with the same ID cause some sort of collision?

File: src/main/java/club/issizler/okyanus/api/cmdnew/BlockCommandSender.java [hunk 6]
```diff
@@ -0,0 +1,15 @@
+package club.issizler.okyanus.api.cmdnew;
+
+import club.issizler.okyanus.api.world.Block;
+import org.jetbrains.annotations.NotNull;
+
+public interface BlockCommandSender extends CommandSender {
```
Comment (Admicos) [inline src/main/java/club/issizler/okyanus/api/cmdnew/BlockCommandSender.java:6]: When would this be used?

**Review by Admicos — COMMENTED** [review]
So far seems OK. But it's kind of big. You probably should've split this into 4 PRs, like JSON, NBT, annotations, and command changes. (Well, this one can stay, just make sure to make smaller PRs next time)

Also, you should have descriptive names for your PRs, just "Develop" doesn't tell what's going on here.

File: src/main/java/club/issizler/okyanus/api/cmdnew/BlockCommandSender.java (hunk 6 above)
Comment (Admicos) [reply src/main/java/club/issizler/okyanus/api/cmdnew/BlockCommandSender.java:6]: Ah, command blocks. Forgot all about them.

File: src/main/java/club/issizler/okyanus/api/json/ClickFeature.java (hunk 3 above)
Comment (portlek) [reply src/main/java/club/issizler/okyanus/api/json/ClickFeature.java:16]: I don't know how to use it. For now, it's working, maybe later I can improve it with the Gson.

File: src/main/java/club/issizler/okyanus/api/json/JsonMessage.java (hunk 4 above)
Comment (portlek) [reply src/main/java/club/issizler/okyanus/api/json/JsonMessage.java:11]: The messages must send own message itself. The opp logic is like that.

File: src/main/java/club/issizler/okyanus/api/json/click/RunConsumer.java (hunk 5 above)
Comment (portlek) [reply src/main/java/club/issizler/okyanus/api/json/click/RunConsumer.java:24]: Actually, no. Our command system is checking just 'label' of the first CommandOf.

**Review by Admicos — APPROVED** [review]
Assuming everything I said before got fixed, the changes look good to me.

Sorry for the delay, btw. I was working on something else.

**Pull request merged**